socorro/processor/rules/mozilla.py

```diff
@@ -879,60 +883,97 @@ class OSPrettyVersionRule(Rule):
-        processed_crash["os_pretty_version"] = pretty_name
+    def parse_version(self, os_version):
+        if not os_version or not isinstance(os_version, str):
+            return None
 
-        os_version = processed_crash.get("os_version") or ""
         match = self.MAJOR_MINOR_RE.match(os_version)
         if match is None:
             # The version number is missing or invalid, there's nothing more to do
-            return True
+            return None
 
         major_version = int(match.group(1))
         minor_version = int(match.group(2))
 
-        os_name = processed_crash.get("os_name") or ""
+        return (major_version, minor_version)
 
-        if os_name.lower().startswith("windows"):
-            if (major_version, minor_version) == (10, 0) and os_version >= "10.0.21996":
-                windows_version = "Windows 11"
+    def compute_windows_pretty_version(self, os_name, os_version, processed_crash):
+        result = self.parse_version(os_version)
+        if result is None:
+            return os_name
 
-            else:
-                windows_version = self.WINDOWS_VERSIONS.get(
-                    "%s.%s" % (major_version, minor_version), "Windows Unknown"
-                )
+        major_version, minor_version = result
 
-            processed_crash["os_pretty_version"] = windows_version
-            return
+        if (major_version, minor_version) == (10, 0) and os_version >= "10.0.21996":
+            pretty_version = "Windows 11"
+
+        else:
+            windows_version = f"{major_version}.{minor_version}"
+            pretty_version = self.WINDOWS_VERSIONS.get(
+                windows_version, "Windows Unknown"
+            )
+
+        return pretty_version
+
+    def compute_macos_pretty_version(self, os_name, os_version, processed_crash):
+        result = self.parse_version(os_version)
+        if result is None:
+            return os_name
+
+        major_version, minor_version = result
+
+        # https://en.wikipedia.org/wiki/MacOS#Release_history
+        if major_version >= 11:
+            # NOTE(willkg): this assumes Apple versions macOS with just the major
+            # version going forward.
+            pretty_version = "macOS %s" % major_version
+        elif major_version >= 10 and minor_version >= 0:
+            pretty_version = "OS X %s.%s" % (major_version, minor_version)
+        else:
+            pretty_version = "OS X Unknown"
+
+        return pretty_version
+
+    def compute_linux_pretty_version(self, processed_crash):
+        pretty_version = glom(
+            processed_crash, "json_dump.lsb_release.description", default=""
+        )
+        pretty_version = pretty_version or "Linux"
+        return pretty_version
+
+    def compute_android_pretty_version(self, os_name, os_version):
+        if os_version:
+            return f"{os_name} {os_version}"
+        return os_name
+
+    def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
```

Comment:
This method was kind of a big mess. I split out the platform-specific code into separate methods to make it much easier to reason about and follow the flow.